eng/versioning.targets

```diff
@@ -40,17 +40,24 @@
   <ItemGroup Condition="('$(TargetsBrowser)' == 'true' or '$(CrossPlatformAndHasNoBrowserTarget)' == 'true') and '$(IsTestProject)' != 'true'">
     <SupportedPlatform Include="browser"/>
   </ItemGroup>
-
-  <!-- Enables warnings for tvOS targeted builds -->
-  <ItemGroup Condition="'$(TargetstvOS)' == 'true' and '$(IsTestProject)' != 'true'">
-    <SupportedPlatform Include="tvos"/>
+  
+  <!-- Add target platforms into MSBuild SupportedPlatform list -->
+  <ItemGroup Condition="'$(TargetsAnyOS)' != 'true' and '$(IsTestProject)' != 'true'">
+    <SupportedPlatform Condition="'$(Targetsillumos)' == 'true'" Include="illumos"/>
+    <SupportedPlatform Condition="'$(TargetsMacCatalyst)' == 'true'" Include="MacCatalyst"/>
+    <SupportedPlatform Condition="'$(TargetsSolaris)' == 'true'" Include="Solaris"/>
+    <SupportedPlatform Condition="'$(TargetstvOS)' == 'true'" Include="tvos"/>
```

Comment:
what about ios?

Reply:
IOS was already included in MSBuild list, recently we removed that from SDK and @jeffhandley  [added into runtime](https://github.com/dotnet/runtime/pull/50143) temporarily, I am not gonna integrate them with this one because we are planning to handle them differently (it might go back to SDK)

Reply:
Ok, should we remove MAcCatalyst from this list then? It seems like in that change MacCatalyst was included as well?